test/general/implementation_details/contains_host_pointers.pass.cpp

```diff
@@ -0,0 +1,192 @@
+        auto all_view = std::ranges::views::all(mr);
+
+        auto zip_view = oneapi::dpl::__ranges::make_zip_view(all_view, all_view);
+
+        static_assert(contains_host_pointer_v<decltype(all_view)> == true);
+        static_assert(contains_host_pointer_v<decltype(zip_view)> == false);
+        static_assert(contains_host_pointer_on_any_layers_v<decltype(zip_view)> == true);
+    }
+
+    // Check that MinimalisticRangeForIntVec can't be used in oneDPL algorithms
+    // as far as it contain host pointer in the std::ranges::ref_view
+    {
+        IntVector vec;
+
+        MinimalisticViewWithSubscription mrv(vec.begin(), vec.end());
+        auto all_view1 = std::ranges::views::all(mrv);
+
+        MinimalisticRangeForIntVec mr(vec.begin(), vec.end());
+        auto all_view2 = std::ranges::views::all(mr);
+
+        auto zip_view = oneapi::dpl::__ranges::make_zip_view(all_view1, all_view2);
+
+        static_assert(contains_host_pointer_v<decltype(all_view1)> == false);
+        static_assert(contains_host_pointer_v<decltype(all_view2)> == true);
+        static_assert(contains_host_pointer_v<decltype(zip_view)> == false);
+        static_assert(contains_host_pointer_on_any_layers_v<decltype(zip_view)> == true);
+    }
+
+    // Check that MinimalisticViewWithSubscription can be used in oneDPL algorithms
+    // as it doesn't contain host pointer in the std::ranges::ref_view
+    {
+        IntVector vec;
+
+        MinimalisticViewWithSubscription mrv(vec.begin(), vec.end());
+        auto all_view = std::ranges::views::all(mrv);
+
+        auto zip_view = oneapi::dpl::__ranges::make_zip_view(all_view, all_view);
+
+        static_assert(contains_host_pointer_v<decltype(all_view)> == false);
+        static_assert(contains_host_pointer_v<decltype(zip_view)> == false);
+        static_assert(contains_host_pointer_on_any_layers_v<decltype(zip_view)> == false);
+    }
+}
+
+// oneapi::dpl::__ranges::__contains_host_pointer functional with std::ranges::take_view
+void
+check_contains_host_pointer_in_take_view()
+{
+    // Check that MinimalisticRangeForIntVec can't be used in oneDPL algorithms
+    // as far as it contain host pointer in the std::ranges::ref_view
+    IntVector vec;
+    MinimalisticRangeForIntVec mr(vec.begin(), vec.end());
+    auto all_view = std::ranges::views::all(mr);
+    auto taken_view = std::ranges::take_view(all_view, all_view.size());
+
+    static_assert(contains_host_pointer_v<decltype(all_view)> == true);
+    static_assert(contains_host_pointer_v<decltype(taken_view)> == false);
+    static_assert(contains_host_pointer_on_any_layers_v<decltype(taken_view)> == true);
+}
+
+// oneapi::dpl::__ranges::__contains_host_pointer functional with std::ranges::drop_view
+void
+check_contains_host_pointer_in_drop_view()
+{
+    // Check that MinimalisticRangeForIntVec can't be used in oneDPL algorithms
+    // as far as it contain host pointer in the std::ranges::ref_view
```

Comment:
A typo: contain -> contains.
Ultimately, I recommend removing these noisy comments or keeping only one - the names of the functions and their short bodies are self-explanatory.

Reply:
Comments has been removed